Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm

```diff
@@ -3298,7 +3299,7 @@ static void elementPositionInformation(WebPage& page, Element& element, const In
 
     if (linkElement && !info.isImageOverlayText) {
         info.isLink = true;
-        info.url = page.applyLinkDecorationFiltering(linkElement->document().completeURL(linkElement->getAttribute(HTMLNames::hrefAttr)), LinkDecorationFilteringTrigger::Unspecified);
+        info.url = page.applyLinkDecorationFiltering(document->completeURL(linkElement->getAttribute(HTMLNames::hrefAttr)), LinkDecorationFilteringTrigger::Unspecified);
```

Comment:
Is this an unrelated fix?

Reply:
Oh, I suppose it's always the case that `linkElement->document() == document` here? I haven't followed the logic right above this.

Reply:
Right — this is just because I pulled `document` out into a smart pointer for use in a few places in this function. `linkElement->document()` is equal to `element.document()`, since `containingLinkAnchorElement` does not cross document boundaries.